lib/Sema/CalleeCandidateInfo.cpp

```diff
@@ -476,10 +471,14 @@ CalleeCandidateInfo::evaluateCloseness(UncurriedCandidate candidate,
       if (matched)
         continue;
       
-      if (archetypesMap.empty())
-        mismatchesAreNearMisses &= argumentMismatchIsNearMiss(argType, paramType);
-      
-      ++mismatchingArgs;
+      // If the real argument is unresolved, the candidate isn't a mismatch because
+      // the type could be anything, but it's still useful to save the argument as
+      // failureInfo.
+      if (!rArgType->hasUnresolvedType()) {
+        if (archetypesMap.empty())
```

Comment:
Nice! You'll soon discover the archetypeMap is a total disaster, BTW :-)

Reply:
Pretty sure it's of my making from generics diagnosis work a year or more ago, so it's karma. :-)